connector/src/main/scala/com/microsoft/kusto/spark/common/KustoOptions.scala

```diff
@@ -25,31 +30,34 @@ trait KustoOptions {
 
   val KUSTO_ACCESS_TOKEN: String = newOption("accessToken")
 
-  // Use only for local runs where one can open the browser
+  // Use only for local runs where one can open the browser - set to true to enable
   val KUSTO_USER_PROMPT: String = newOption("userPrompt")
   // AAD application pfx certificate path
   val KUSTO_AAD_APP_CERTIFICATE_PATH: String = newOption("kutoAadAppCertPath")
   // AAD application certificate password
   val KUSTO_AAD_APP_CERTIFICATE_PASSWORD: String = newOption("kutoAadAppCertPassword")
 
+  // KeyVault options. Relevant only if credentials need to be retrieved from Key Vault
+  val KEY_VAULT_URI = "keyVaultUri"
+  val KEY_VAULT_APP_ID = "keyVaultAppId"
+  val KEY_VAULT_APP_KEY = "keyVaultAppKey"
+
   // Classpath to a class that its constructor accepts one argument of type CaseInsensitiveMap[String] which will contain
   // the options provided to the connector. The class should extend Callable[String] with Serializeable and is expected
   // to return an AAD token upon invoking the call method.
   // The provider will be called for every request to the kusto service
   val KUSTO_TOKEN_PROVIDER_CALLBACK_CLASSPATH: String = newOption("tokenProviderCallbackClasspath")
 
-  // Target/source Kusto cluster for writing/reading the data.
-  val KUSTO_CLUSTER: String = newOption("kustoCluster")
-  // Target/source Kusto database for writing/reading the data. See KustoSink.md/KustoSource.md for
-  // required permissions
-  val KUSTO_DATABASE: String = newOption("kustoDatabase")
+  /** Optional parameters */
+  // Kusto ingestion uri cluster for reading data - provide this if ingestion uri cannot be deducted from adding
```

Comment:
"Kusto ingestion cluster URL for reading data - provide this if ingestion URL cannot be deduced from adding"